packages/jest/src/plugins/plugin.ts

```diff
@@ -208,90 +210,173 @@ async function buildJestTargets(
-        : await source.getTestPaths(config.globalConfig);
-
-    const testPaths = new Set(specs.tests.map(({ path }) => path));
-
-    if (testPaths.size > 0) {
-      const groupName = 'E2E (CI)';
-      const targetGroup = [];
-      metadata = {
-        targetGroups: {
-          [groupName]: targetGroup,
-        },
-      };
-      const dependsOn: string[] = [];
-
-      targets[options.ciTargetName] = {
-        executor: 'nx:noop',
-        cache: true,
-        inputs,
-        outputs,
-        dependsOn,
-        metadata: {
-          technologies: ['jest'],
-          description: 'Run Jest Tests in CI',
-          nonAtomizedTarget: options.targetName,
-          help: {
-            command: `${pmc.exec} jest --help`,
-            example: {
-              options: {
-                coverage: true,
+      rawConfig.coverageDirectory
+        ? join(context.workspaceRoot, projectRoot, rawConfig.coverageDirectory)
+        : undefined,
+      undefined,
+      context
+    ));
+
+    if (options?.ciTargetName) {
+      const testPaths = await getTestPaths(
+        projectRoot,
+        rawConfig,
+        absConfigFilePath,
+        context,
+        presetCache
+      );
+
+      if (testPaths.length > 0) {
+        const targetGroup = [];
+        metadata = {
+          targetGroups: {
+            [groupName]: targetGroup,
+          },
+        };
+        const dependsOn: string[] = [];
+
+        targets[options.ciTargetName] = {
+          executor: 'nx:noop',
+          cache: true,
+          inputs,
+          outputs,
+          dependsOn,
+          metadata: {
+            technologies: ['jest'],
+            description: 'Run Jest Tests in CI',
+            nonAtomizedTarget: options.targetName,
+            help: {
+              command: `${pmc.exec} jest --help`,
+              example: {
+                options: {
+                  coverage: true,
+                },
               },
             },
           },
-        },
-      };
-      targetGroup.push(options.ciTargetName);
-
-      for (const testPath of testPaths) {
-        const relativePath = normalizePath(
-          relative(join(context.workspaceRoot, projectRoot), testPath)
-        );
-        const targetName = `${options.ciTargetName}--${relativePath}`;
-        dependsOn.push(targetName);
-        targets[targetName] = {
-          command: `jest ${relativePath}`,
-          cache,
+        };
+        targetGroup.push(options.ciTargetName);
+
+        const specIgnoreRegexes: undefined | RegExp[] =
+          rawConfig.testPathIgnorePatterns?.map(
+            (p: string) => new RegExp(replaceRootDirInPath(projectRoot, p))
+          );
+
+        for (const testPath of testPaths) {
+          const relativePath = normalizePath(
+            relative(join(context.workspaceRoot, projectRoot), testPath)
+          );
+          if (specIgnoreRegexes?.some((regex) => regex.test(relativePath)))
```

Comment:
It would be more correct to move the check for ignored specs into the `getTestPaths` function or at least to filter the ignored specs before the [line 242](https://github.com/nrwl/nx/pull/28522/files#diff-a1531bb2dca28c044b3dc539f643224c0f946c60208830a3bc91402eb50abeaaR242). It's definitely an edge case and wouldn't break anything, but the way it is right now, it _**could**_ happen that we create the `e2e-ci` target, but then, we don't create any target for any specs because they are all ignored.

As mentioned, it's not exactly an issue, as in nothing would break, but it wouldn't match the plugin's outcome when using the `jest-runtime` package.

Reply:
Changed the check to ensure that `targetGroup` is not empty, so that we do not create `e2e-ci` target unless there is at least one spec that that in the `targetGroup`.